ethereum/statetransition/src/main/java/tech/pegasys/teku/statetransition/datacolumns/DasSamplerBasic.java

```diff
@@ -76,23 +82,27 @@ public SafeFuture<List<DataColumnSidecar>> checkDataAvailability(
     return SafeFuture.collectAll(
             columnIndexes.get().stream()
                 .map(
-                    index ->
-                        custody.getCustodyDataColumnSidecar(
-                            new DataColumnIdentifier(blockRoot, index))))
-        .thenApply(
-            listOfOptionals -> {
-              // TODO: remove or move to debug logging
-              LOG.info(
-                  "Collected {} DataColumnSidecars in checkDataAvailability for {}({})",
-                  listOfOptionals.stream()
-                      .map(
-                          maybeSidecar ->
-                              maybeSidecar.map(DataColumnSidecar::toLogString).orElse("[empty]"))
-                      .collect(Collectors.joining(",")),
-                  blockRoot,
-                  slot);
-              return listOfOptionals.stream().map(Optional::orElseThrow).toList();
-            });
+                    index -> getLocallyOrRetrieve(new DataColumnSlotAndIdentifier(slot, new DataColumnIdentifier(blockRoot, index)))))
+        .thenPeek(
+            // TODO: remove or move to debug logging
+            sidecars -> LOG.info(
```

Comment:
I don't understand this change. Before it was clear which sidecars are missed, which helps investigating why it happens, after change it will be not possible. The log line before the change in case of failure example: 
```
2024-09-10 22:21:33.110 INFO  - Collected [empty],[empty],[empty],[empty],[empty],[empty],[empty],[empty],[empty],[empty],DataColumnSidecar[block 8987e8..041e (10), index 10, 1st cell 00e4eab2c25b0c, commitments 3, proofs 3],DataColumnSidecar[block 8987e8..041e (10), index 11, 1st cell 00419b1467e9df, commitments 3, proofs 3],DataColumnSidecar[block 8987e8..041e (10), index 12, 1st cell 00000000000000, commitments 3, proofs 3],DataColumnSidecar[block 8987e8..041e (10), index 13, 1st cell 00000000000000, commitments 3, proofs 3],DataColumnSidecar[block 8987e8..041e (10), index 14, 1st cell 00000000000000, commitments 3, proofs 3],DataColumnSidecar[block 8987e8..041e (10), index 15, 1st cell 00000000000000, commitments 3, proofs 3],DataColumnSidecar[block 8987e8..041e (10), index 16, 1st cell 00000000000000, commitments 3, proofs 3],DataColumnSidecar[block 8987e8..041e (10), index 17, 1st cell 00000000000000, commitments 3, proofs 3],DataColumnSidecar[block 8987e8..041e (10), index 18, 1st cell 00000000000000, commitments 3, proofs 3],DataColumnSidecar[block 8987e8..041e (10), index 19, 1st cell 00000000000000, commitments 3, proofs 3],DataColumnSidecar[block 8987e8..041e (10), index 20, 1st cell 00000000000000, commitments 3, proofs 3],DataColumnSidecar[block 8987e8..041e (10), index 21, 1st cell 00000000000000, commitments 3, proofs 3],DataColumnSidecar[block 8987e8..041e (10), index 22, 1st cell 00000000000000, commitments 3, proofs 3],DataColumnSidecar[block 8987e8..041e (10), index 23, 1st cell 00000000000000, commitments 3, proofs 3],DataColumnSidecar[block 8987e8..041e (10), index 24, 1st cell 00000000000000, commitments 3, proofs 3],DataColumnSidecar[block 8987e8..041e (10), index 25, 1st cell 00000000000000, commitments 3, proofs 3],DataColumnSidecar[block 8987e8..041e (10), index 26, 1st cell 00000000000000, commitments 3, proofs 3],DataColumnSidecar[block 8987e8..041e (10), index 27, 1st cell 00000000000000, commitments 3, proofs 3],DataColumnSidecar[block 8987e8..041e (10), index 28, 1st cell 00000000000000, commitments 3, proofs 3],DataColumnSidecar[block 8987e8..041e (10), index 29, 1st cell 00000000000000, commitments 3, proofs 3],DataColumnSidecar[block 8987e8..041e (10), index 30, 1st cell 00000000000000, commitments 3, proofs 3],DataColumnSidecar[block 8987e8..041e (10), index 31, 1st cell 00000000000000, commitments 3, proofs 3],DataColumnSidecar[block 8987e8..041e (10), index 32, 1st cell 00000000000000, commitments 3, proofs 3],DataColumnSidecar[block 8987e8..041e (10), index 33, 1st cell 00000000000000, commitments 3, proofs 3],DataColumnSidecar[block 8987e8..041e (10), index 34, 1st cell 00000000000000, commitments 3, proofs 3],DataColumnSidecar[block 8987e8..041e (10), index 35, 1st cell 00000000000000, commitments 3, proofs 3],DataColumnSidecar[block 8987e8..041e (10), index 36, 1st cell 00000000000000, commitments 3, proofs 3],DataColumnSidecar[block 8987e8..041e (10), index 37, 1st cell 00000000000000, commitments 3, proofs 3],DataColumnSidecar[block 8987e8..041e (10), index 38, 1st cell 00000000000000, commitments 3, proofs 3],DataColumnSidecar[block 8987e8..041e (10), index 39, 1st cell 00000000000000, commitments 3, proofs 3],DataColumnSidecar[block 8987e8..041e (10), index 40, 1st cell 00000000000000, commitments 3, proofs 3],DataColumnSidecar[block 8987e8..041e (10), index 41, 1st cell 00000000000000, commitments 3, proofs 3],DataColumnSidecar[block 8987e8..041e (10), index 42, 1st cell 00000000000000, commitments 3, proofs 3],DataColumnSidecar[block 8987e8..041e (10), index 43, 1st cell 00000000000000, commitments 3, proofs 3],DataColumnSidecar[block 8987e8..041e (10), index 44, 1st cell 00000000000000, commitments 3, proofs 3],DataColumnSidecar[block 8987e8..041e (10), index 45, 1st cell 00000000000000, commitments 3, proofs 3],DataColumnSidecar[block 8987e8..041e (10), index 46, 1st cell 00000000000000, commitments 3, proofs 3],DataColumnSidecar[block 8987e8..041e (10), index 47, 1st cell 00000000000000, commitments 3, proofs 3],DataColumnSidecar[block 8987e8..041e (10), index 48, 1st cell 00000000000000, commitments 3, proofs 3],DataColumnSidecar[block 8987e8..041e (10), index 49, 1st cell 00000000000000, commitments 3, proofs 3],DataColumnSidecar[block 8987e8..041e (10), index 50, 1st cell 00000000000000, commitments 3, proofs 3],DataColumnSidecar[block 8987e8..041e (10), index 51, 1st cell 00000000000000, commitments 3, proofs 3],DataColumnSidecar[block 8987e8..041e (10), index 52, 1st cell 00000000000000, commitments 3, proofs 3],DataColumnSidecar[block 8987e8..041e (10), index 53, 1st cell 00000000000000, commitments 3, proofs 3],DataColumnSidecar[block 8987e8..041e (10), index 54, 1st cell 00000000000000, commitments 3, proofs 3],DataColumnSidecar[block 8987e8..041e (10), index 55, 1st cell 00000000000000, commitments 3, proofs 3],DataColumnSidecar[block 8987e8..041e (10), index 56, 1st cell 00000000000000, commitments 3, proofs 3],DataColumnSidecar[block 8987e8..041e (10), index 57, 1st cell 00000000000000, commitments 3, proofs 3],DataColumnSidecar[block 8987e8..041e (10), index 58, 1st cell 00000000000000, commitments 3, proofs 3],DataColumnSidecar[block 8987e8..041e (10), index 59, 1st cell 00000000000000, commitments 3, proofs 3],DataColumnSidecar[block 8987e8..041e (10), index 60, 1st cell 00000000000000, commitments 3, proofs 3],DataColumnSidecar[block 8987e8..041e (10), index 61, 1st cell 00000000000000, commitments 3, proofs 3],DataColumnSidecar[block 8987e8..041e (10), index 62, 1st cell 00000000000000, commitments 3, proofs 3],DataColumnSidecar[block 8987e8..041e (10), index 63, 1st cell 00000000000000, commitments 3, proofs 3],DataColumnSidecar[block 8987e8..041e (10), index 64, 1st cell 713522cde5b292, commitments 3, proofs 3],DataColumnSidecar[block 8987e8..041e (10), index 65, 1st cell 502651a7468cad, commitments 3, proofs 3],DataColumnSidecar[block 8987e8..041e (10), index 66, 1st cell 2dba6b488180f4, commitments 3, proofs 3],DataColumnSidecar[block 8987e8..041e (10), index 67, 1st cell 029c2aa3a7d15a, commitments 3, proofs 3],DataColumnSidecar[block 8987e8..041e (10), index 68, 1st cell 4c1ce12f039cdf, commitments 3, proofs 3],DataColumnSidecar[block 8987e8..041e (10), index 69, 1st cell 18f1bc01a32d79, commitments 3, proofs 3],DataColumnSidecar[block 8987e8..041e (10), index 70, 1st cell 41d0a5682af0ed, commitments 3, proofs 3],DataColumnSidecar[block 8987e8..041e (10), index 71, 1st cell 6f0119d080ff4d, commitments 3, proofs 3],DataColumnSidecar[block 8987e8..041e (10), index 72, 1st cell 2c334986daa353, commitments 3, proofs 3],DataColumnSidecar[block 8987e8..041e (10), index 73, 1st cell 3eea33e23f140b, commitments 3, proofs 3],DataColumnSidecar[block 8987e8..041e (10), index 74, 1st cell 5c68cc3d477736, commitments 3, proofs 3],DataColumnSidecar[block 8987e8..041e (10), index 75, 1st cell 183ba2139a066c, commitments 3, proofs 3],DataColumnSidecar[block 8987e8..041e (10), index 76, 1st cell 5c36fdd45b9234, commitments 3, proofs 3],DataColumnSidecar[block 8987e8..041e (10), index 77, 1st cell 36608275601e65, commitments 3, proofs 3],DataColumnSidecar[block 8987e8..041e (10), index 78, 1st cell 5c5e448b099d0c, commitments 3, proofs 3],DataColumnSidecar[block 8987e8..041e (10), index 79, 1st cell 5fae3338bcbd94, commitments 3, proofs 3],DataColumnSidecar[block 8987e8..041e (10), index 80, 1st cell 15f3e000ecb034, commitments 3, proofs 3],[empty],[empty],[empty],[empty],[empty],[empty],[empty],[empty],[empty],[empty],[empty],[empty],[empty],[empty],[empty],[empty],[empty],[empty],[empty],[empty],[empty],[empty],[empty],[empty],[empty],[empty],[empty],[empty],[empty],[empty],[empty],[empty],[empty],[empty],[empty],[empty],[empty],[empty],[empty],[empty],[empty],[empty],[empty],[empty],[empty],[empty],[empty] DataColumnSidecars in checkDataAvailability for 0x8987e84c311197f5cae68a53183e9841d72c6edbc7b8aee2239282914cdc041e(10)

```

Reply:
Yeah, it makes sense to make the log more descriptive. Let me think about it...